tasks/e2e.rb

```diff
@@ -159,20 +162,20 @@ def self.dump_podfile(config, source)
         end
     end
 
-    def self.build_project(workspace, project, scheme, extra_args = {})
+    def self.build_project(workspace, project, scheme, sdk = 'iphone', platform = 'iOS', extra_args = {})
         xcodebuild_args = {
             'workspace' => workspace,
             'project' => project,
             'scheme' => scheme,
             'configuration' => 'Debug',
-            'sdk' => 'iphonesimulator',
-            'destination' => 'generic/platform=iOS Simulator',
+            'sdk' => "#{sdk}simulator",
+            'destination' => "generic/platform=#{platform} Simulator",
             'derivedDataPath' => DERIVED_DATA_PATH,
         }.merge(extra_args).compact
         xcodebuild_vars = {
-            'EXCLUDED_ARCHS' => 'arm64 i386'
+            'EXCLUDED_ARCHS' => 'arm64'
         }
-        args = ['xcodebuild']
+        args = ['set -o pipefail;', 'xcodebuild']
```

Comment:
errors in `xcodebuild` were not failing a step as `args.push("| tee #{LOG_NAME}")` was swallowing an error.